src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -303,6 +304,36 @@ export default class DataTable extends Vue {
     this.$emit(DATA_TABLE_EVENTS.SELECTED_ROWS_CHANGE, selectedRowsData);
   }
 
+  _checkIndeterminate(rowData: DataTableRow) {
+    let parent = rowData.parent;
+
+    while (parent) {
+      const parentCheck = document.querySelector(
+        `#checkbox-select-${this._rowId(parent.rowNumber)}`
+      ) as HTMLInputElement;
+      const children = parent.nestedContent.table.data;
+      const childrenSelected: DataTableRow[] = [];
+
+      children?.forEach((row: DataTableRow) => {
```

Comment:
It shouldn't be necessary to check all the children..at the moment there is just one selected, the indeterminate class should be applied...except if all the children are selected, but I think you take care of that at line 334, right?